assets/js/components/Dashboard/PredictionSuppressionPage.tsx

```diff
@@ -50,26 +70,93 @@ const PredictionSuppressionPage = () => {
     }, places),
     line,
   );
+
+  const mainLine = fp.startsWith("Green", line) ? "Green" : line;
+
   const lineStopLookup = fp.flow([
     fp.filter({ line: line.split("-")[0] }),
     fp.map(({ stop_id, direction_id, type }) => [
-      lookupKey(stop_id, direction_id),
+      lookupKey([stop_id, direction_id]),
       type,
     ]),
     fp.fromPairs,
   ])(lineStops);
 
+  const suppressedPredictionsLookup = fp.flow([
+    fp.map((p: SuppressedPrediction) => [
+      lookupKey([p.location_id, p.route_id, p.direction_id]),
+      p,
+    ]),
+    fp.fromPairs,
+  ])(suppressedPredictions);
+
   const noPredictions = (place: Place) => {
-    const serviceType = lineStopLookup[lookupKey(place.id, 0)];
+    const serviceType = lineStopLookup[lookupKey([place.id, 0])];
     return (
       line.startsWith("Green") && fp.includes(serviceType, ["start", "end"])
     );
   };
 
-  const renderInput = (place: Place, directionId: number) => {
-    const serviceType = lineStopLookup[lookupKey(place.id, directionId)];
-    const suppressed = false;
-    const expire = false;
+  const renderInput = (
+    locationId: string,
+    stopId: string,
+    directionId: number,
+  ) => {
+    const serviceType = lineStopLookup[lookupKey([stopId, directionId])];
+    const record =
+      suppressedPredictionsLookup[
+        lookupKey([locationId, mainLine, directionId])
+      ];
+
+    const toggleSuppression = () => {
+      if (!suppressedPredictions) {
+        return;
+      }
+      if (record) {
+        const newData = fp.without([record], suppressedPredictions);
+        mutateSuppressedPredictions(
+          async () => {
+            await deleteSuppressedPrediction(record);
+            return newData;
+          },
+          { optimisticData: newData },
+        );
+      } else {
+        const newRecord = {
+          location_id: locationId,
+          route_id: mainLine,
+          direction_id: directionId,
+          clear_at_end_of_day: false,
+        };
+        const newData = fp.concat(suppressedPredictions, [newRecord]);
+        mutateSuppressedPredictions(
+          async () => {
+            await createSuppressedPrediction(newRecord);
+            return newData;
+          },
+          { optimisticData: newData },
+        );
+      }
+    };
+
+    const toggleExpiration = () => {
+      if (!suppressedPredictions) {
+        return;
+      }
+      const newRecord = fp.update(["clear_at_end_of_day"], (v) => !v, record);
+      const newData = fp.map(
+        (p) => (p === record ? newRecord : p),
+        suppressedPredictions,
+      );
+      mutateSuppressedPredictions(
+        async () => {
+          await updateSuppressedPrediction(newRecord);
+          return newData;
+        },
+        { optimisticData: newData },
+      );
+    };
+
     if (serviceType === "start" || serviceType === "mid") {
       const predictionsText =
         serviceType == "start" ? "terminal predictions" : "predictions";
```

Comment:
Super nit since it wasn't even changed in this PR, but I noticed this check is using `==`. Should we use `===` instead to stay in-line with best practice?

Reply:
Thanks, good catch. This is something we can have eslint enforce, so I'll get that set up.